### DISCUSSION TITLE
A few Flows limitations I found annoying

### DISCUSSION THREAD

**Discussion by mnajborowski** [postmanlabs/postman-flows#252] [Ideas]
Hi, I've been using Flows for 2 weeks now and I'd like to share some thoughts not refering to any particular topic:
1. There's no a fast way to change environment of all of the blocks in a flow. Instead you have to select environment dropdown and change it there in each block one by one. I thought I could be smart and create a duplicate of a flow just so it runs in different environment, but that's not possible too :(
2. There's no undo/redo for actions made on a flow. (I've seen there's a lot of request for adding this already :) )
3. There's no spacing grid. Also there's no magnet that could allow you to space the blocks evenly, with straight connection lines.
4. There's no warning when deleting a flow and you cannot undo that action.
5. Pre-request script and Tests don't work for requests called from a flow. This is a huge pain, as there's no way for overwriting current value of environment/global variables. If you have a value that's returned from some request e.g. bearer token, and that token is then passed in each further request, there's no other way than saving that token in a Durables block and then manually assigning its value in each request block. That's a lot of manual work.
6. You can't pass environment variable as a value in Durable block.
7. When assigning a string value in a requetst block or Durables block, the `"` appear on the sides of the value. That's missleading, as one could expect that the string value will include those `"` when passed to a variable used in a request, leading to a type error when used e.g. in a JSON body.

Anyway keep up with the good work! I really see potential in Flows, especially for people that aren't really familiar with JS. 👍

**Comment by saswatds** [comment]
Hey @mnajborowski, thanks for this awesome feedback
1. We completely understand this problem, here is a solution we are looking towards implementing - We identify the blocks the environment is used in and provide a way to change in all.
2. Are you on the latest version of Flow? I have actually implemented the undo and redo feature. Ctrl+Z and Ctrl+Shift+Z have been mapped to undo and redo.
3. We are soon adding this feature. We want it too. 
4. Could you explain a bit as to when this is happening. Deleting a flow from the sidebar should ideally show a modal to confirm
5. This is good feedback. I think we could improve the Request block to automatically save it to durable. 
6. Would it be helpful if I add the ability to access `Global Variables` in the durables block? 
7. @samay-v This is correct. Let's take a review on the representation of the values again.

**Reply by mnajborowski** [reply]
1. 👍 
2. Yeah, I must have been on older version. Undo/redo works like a charm now :)
3. 👍 
4. You're right, I was wrong about that.
5. If by "improve the Request block to automatically save it to durable" you mean overwriting current Globals/Environment variable values, than it's just what I need :)
6. Yes, that is just what I need!
7. 👍
